scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2878,252 @@ cdef class Rotation:
+        if weights.ndim != 1:
+            raise ValueError("Expected `weights` to be 1 dimensional, got "
+                             "shape {}.".format(weights.shape))
+        if N > 1 and (weights.shape[0] != b.shape[0]):
+            raise ValueError("Expected `weights` to have number of values"
+                             "equal to number of input vectors, got "
+                             "{} values and {} vectors.".format(
+                                weights.shape[0], b.shape[0]))
+        if (weights < 0).any():
+            raise ValueError("`weights` may not contain negative values")
+
+        # Check for an infinite weight, which indicates that the corresponding
+        # vector pair is the primary unmoving reference which we align the
+        # other vectors to.
+        weight_inf = np.isposinf(weights)
+        if sum(weight_inf) > 1:
+            raise ValueError("Only one infinite weight is allowed")
+        elif N >= 2 and weight_inf.any():
+            if N == 2:
+                # A single non-infinite weight vector only needs normalization
+                a_non_inf = a[~weight_inf].reshape(3)
+                b_non_inf = b[~weight_inf].reshape(3)
+            else:  # N > 2
+                # Multiple non-infinite weight vectors need to be best aligned
+                R, _ = cls.align_vectors(a[~weight_inf], b[~weight_inf],
+                                         weights=weights[~weight_inf])
+                R_mat = R.as_matrix()
+                a_non_inf = R_mat[0, :]  # Already normalized
+                b_non_inf = R_mat[1, :]
+
+            # Reshape to 1D vectors
+            a_inf = a[weight_inf].reshape(3)
+            b_inf = b[weight_inf].reshape(3)
+
+            # Calculate angles between the primary and secondary vectors
+            a_angle = acos(np.dot(a_inf, a_non_inf))
+            b_angle = acos(np.dot(b_inf, b_non_inf))
+
+            # If the primary and secondary vectors are not too close to
+            # have numerical stability issues, fit orthonormal frames and
+            # find the rotation between them
+            if a_angle >= atol and b_angle >= atol:
+                A = np.array(_matrix_from_pri_sec_vectors(a[0], a[1]))
+                B = np.array(_matrix_from_pri_sec_vectors(b[0], b[1]))
+                R_opt = cls.from_matrix(A.T @ B)
+
+                # Note: we apply to the original vectors, since if N > 2 and we
+                # best fit the non-infinite weight vectors, we need to
+                # calculate rssd on the full set. The error will be 0 for the
+                # infinite weight vector.
+                weights_non_inf = weights.reshape(len(weights), 1)
+                weights_non_inf[weight_inf] = 0
+                a_est = R_opt.apply(b)
+                rssd = np.sqrt(np.sum(weights_non_inf * (a_est - a)**2))
+
+                if return_sensitivity:
+                    # Note: the sensitivity is not meaningful with an infinite
```

Comment:
I don't think it's "meaningful" per se. But I don't quite understand how to compute it here.

Looks like there should be 2 sensitivities: for the "strict" vectors and for other vectors. Like "how the rotation matrix changes, if we perturb the strict vectors" and "how the rotation matrix changes, if perturb other vectors".

Reply:
As for the handling. What if we raise an error when `sensitivity=True` and an infinite weight presents?

Could it be better than returning infinities? I'm not sure, think about it.

Reply:
My goal with the behavior of the code path with an infinite weight, is that it would return the same result as `weights = [w0, w1, w2, ...]` in the limit of `w0 -> inf`. With the existing algorithm, I see that as I increase one of the weights towards infinity, the entries in the sensitivity matrix also approach infinity. So I think this accurately captures the behavior in the limit. We could also raise an error, what do you think?

I think your comment about the sensitivities for the strict vs other vectors is a good idea, but I'm not sure how to rigorously think about it.

Reply:
I think it's possible to compute these 2 sensitivities, but I don't like the idea of changing the number of returned objects once more. (I think it was a bad pattern to choose here in the first place).

I would probably start with an exception for this case, because we don't know what to return.

Reply:
Sounds good to me, changed this to throw an exception.